### PR TITLE
KRACOEUS-8839:Fix up commit related to attachments

### PR DESCRIPTION
**Pull request by blackcathacker**
Rename base class to match up with attachment dao given other different usages for AttachmentDataSource.java.
Add javadoc and other clean up.
Code review changes from #140

### CODE REVIEW

File: coeus-impl/src/main/java/org/kuali/coeus/common/framework/print/AttachmentDataSource.java [hunk 1]
```diff
@@ -31,19 +31,12 @@
  * This class helps create the foundation of attachment data sources.
  */
 @MappedSuperclass
```
Comment (leoherbie) [inline coeus-impl/src/main/java/org/kuali/coeus/common/framework/print/AttachmentDataSource.java]: I'm guessing this thing should no longer be a mapped super class

Comment (leoherbie) [inline coeus-impl/src/main/java/org/kuali/coeus/common/framework/print/AttachmentDataSource.java]: remove import javax.persistence.*;

**Comment by leoherbie** [comment]
done reviewing.

File: coeus-impl/src/main/java/org/kuali/coeus/common/framework/print/AttachmentDataSource.java (hunk 1 above)
Comment (blackcathacker) [inline coeus-impl/src/main/java/org/kuali/coeus/common/framework/print/AttachmentDataSource.java]: Fixed.

**Comment by leoherbie** [comment]
yes please make it Serializable just in case

**Pull request merged**